### DISCUSSION TITLE
Frontend for Wakapi And A Hard Fork

### DISCUSSION THREAD

**Discussion by jemiluv8** [muety/wakapi#714] [General]
### NEXTJS WAKAPI FRONTEND

I've created a nextjs frontend for wakapi. It is available in my [fork](https://github.com/jemiluv8/wakana/tree/monorepo).
Make sure you're under the mono repo branch.

### NEW ENV VAR
- All you need is to set the jwt_secret in your yml config 

### FRONTEND ENV
All you need is in the .env.local
I implemented github login but the original email password login/signup still works.

### THE FORK AND MORE
Based on my analysis of the github issues over the past few months, 
a move to something like nextjs didn't seem on the cards. Plus, I was 
curious to see how this will come together. I'm glad it did.

I had commercial aspirations for this and still do but my day job is starting to get more
interesting and I just don't have the resources just yet. I can't wait to see how you fine
people do with your self-hosting and stuff. Good luck.

### Feature Requests and Github Stars
- You can create issues in the forked repository but I might not work on them. Sorry.
- And please star the repository if you find this useful

<img width="1436" alt="Screenshot 2024-11-23 at 6 34 39 PM" src="https://github.com/user-attachments/assets/c2d3075d-7668-4b83-947a-28d4eab4cba4">

<img width="1437" alt="Screenshot 2024-11-22 at 11 57 06 AM" src="https://github.com/user-attachments/assets/81d81ca3-af22-43a4-9a68-ca4afc075824">

<img width="1435" alt="Screenshot 2024-11-22 at 11 56 52 AM" src="https://github.com/user-attachments/assets/e9cb2439-7c63-4422-a1be-e28312a7ced7">

**Comment by alanhamlett** [comment]
Looks like you just ripped off the official [WakaTime](https://wakatime.com) website, and then you plan to charge money for it? WTF

**Reply by jemiluv8** [reply]
Hello @alanhamlett. Thanks for the amazing Wakatime product. I still use it by the way.

### About your WTF
I know how it looks and for all the great work your company has done with the plugins, I can understand your
sentiment. And of course, I'm no designer and took inspiration from your website. But FYI this is now open source
and in the public domain. I haven't commercialised it yet. And probably won't. 

If that changes, I might reach out. In the meantime, it is just a hobby project.
